File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        int least = INT_MIN;
        stack<int> st;

        for(int i = nums.size() - 1; i >= 0; i--) {
            if(nums[i] < least) return true;
            else while(!st.empty() && nums[i] > st.top()) {
                least = st.top();
                st.pop();
            }
            st.push(nums[i]);
        }

        return false;
    }
};

File: 0456-132-pattern/0456-132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        least = float('-inf')
        st = []

        for i in range(len(nums) - 1, -1, -1):
            if nums[i] < least:
                return True
            while st and nums[i] > st[-1]:
                least = st.pop()
            st.append(nums[i])

        return False